github/codeowners.py

```diff
@@ -12,21 +12,51 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import collections.abc
+import dataclasses
 from pathlib import Path
 import logging
 import typing
 
-from github3 import GitHub
+from github3 import GitHub, GitHubEnterprise
 from github3.exceptions import NotFoundError
+import github3.exceptions
+import github3.orgs
 import github3.repos.repo
+import github3.search.user
+import github3.users
 
-from ci.util import existing_dir, existing_file, not_none
+from ci.util import existing_dir, existing_file
 import ci.log
 
 logger = logging.getLogger(__name__)
 ci.log.configure_default_logging()
 
 
+@dataclasses.dataclass(frozen=True)
```

Comment:
as already discussed earlier, I do not see any point in using a dataclass here, and also do not see a lot of benefits from having a common base class

- inherit from str
- consider removing base-class

if keeping base-class, we might consider to add parsing to it (not sure whether this would result in nice code, though)